lib/team_statistics.rb

```diff
@@ -106,4 +106,142 @@ def fewest_goals_scored(team_id)
       game.goals
     end.goals
   end
+
+  def favorite_opponent(team_id)
+    team_id = team_id.to_i
+
+    filtered_games = []
+    @games.each do |game_id, game|
+      if game.home_team_id == team_id || game.away_team_id == team_id
+        filtered_games.push(game)
+      end
+    end
```

Comment:
We could use a helper method for this:
```
filtered_games = []
   @games.each do |game_id, game|
      if game.home_team_id == team_id || game.away_team_id == team_id
        filtered_games.push(game)
   end
end
```
since we use it 4 different times.